src/machine/machine_atsamd21.go

```diff
@@ -18,113 +18,93 @@ import (
 
 const CPU_FREQUENCY = 48000000
 
-type GPIOMode uint8
+type PinMode uint8
 
 const (
-	GPIO_ANALOG         = 1
-	GPIO_SERCOM         = 2
-	GPIO_SERCOM_ALT     = 3
-	GPIO_TIMER          = 4
-	GPIO_TIMER_ALT      = 5
-	GPIO_COM            = 6
-	GPIO_AC_CLK         = 7
-	GPIO_DIGITAL        = 8
-	GPIO_INPUT          = 9
-	GPIO_INPUT_PULLUP   = 10
-	GPIO_OUTPUT         = 11
-	GPIO_PWM            = GPIO_TIMER
-	GPIO_PWM_ALT        = GPIO_TIMER_ALT
-	GPIO_INPUT_PULLDOWN = 12
+	PinAnalog    PinMode = 1
+	PinSerCom    PinMode = 2
+	PinSerComAlt PinMode = 3
```

Comment:
Probably should name this `PinSERCOMAlt` only because the datasheet and all other all use all upper case for the `SERCOM`.